cmd/collector/main.go

```diff
@@ -48,10 +51,30 @@ const (
 )
 
 func main() {
+	serverState := http.StatusServiceUnavailable
+
+	logger, _ := zap.NewProduction()
+	var serverChannel = make(chan os.Signal, 0)
+	signal.Notify(serverChannel, os.Interrupt, syscall.SIGTERM)
+
+	go func() {
+		// Start the health check server, used for reporting the liveness and readiness state for this component
+		logger.Info("Health Check server started", zap.Int("http-port", *builder.HealthCheckHTTPPort))
+		healthCheckMux := http.NewServeMux()
+		healthCheckMux.HandleFunc("/", func(w http.ResponseWriter, _ *http.Request) {
```

Comment:
For this initial PR, this might be OK, but it would probably be better in a next iteration to extract this into its own source file.